### PR TITLE
Add full CRUD REST API for configuration files + JUnit tests

### PR DESCRIPTION
**Pull request by NowayLM**
### Description
This PR introduces a complete set of REST endpoints to manage YAML configuration files in MetricsHub Agent.
It also adds comprehensive unit tests for both the controller and the service layer.

### Changes included
New REST endpoints `ConfigurationFilesController`
- `GET /api/config-files`
Returns a list of all configuration files with metadata (name, size, lastModificationTime).

- `GET /api/config-files/content?name=<file>`
Returns the raw content of a specific configuration file as plain text.

- `PUT /api/config-files/content?name=<file>`
Creates or updates a configuration file with the provided YAML content.

- `POST /api/config-files/content?name=<file>`
Validates the YAML syntax of a configuration file.
Returns { "valid": true } or detailed error messages.

- `DELETE /api/config-files/content?name=<file>`
Deletes the specified configuration file.

- `PATCH /api/config-files/content?name=<file>`
Renames a configuration file.
Request body example:
`{ "newName": "new.yaml" }`

Service logic (ConfigurationFilesService)

Unit tests

- Controller tests: ConfigurationFilesControllerTest

- Happy paths and error cases for all endpoints.

- Uses MockMvc to simulate HTTP requests.

- Achieved full controller coverage (10 tests).

- Service tests: ConfigurationFilesServiceTest

- Validates core file operations with a real temp filesystem (@TempDir).

- Tests for edge cases like missing files, invalid YAML, conflicts, etc.

### CODE REVIEW

File: metricshub-agent/src/main/java/org/metricshub/web/service/ConfigurationFilesService.java [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+			// Write to a temp file, then move atomically
+			final Path tmp = Files.createTempFile(dir, fileName + ".", ".tmp");
+			Files.writeString(
+				tmp,
+				content == null ? "" : content,
+				StandardCharsets.UTF_8,
+				StandardOpenOption.TRUNCATE_EXISTING
+			);
+			try {
+				Files.move(tmp, target, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
```
Comment (bertysentry) [inline metricshub-agent/src/main/java/org/metricshub/web/service/ConfigurationFilesService.java:146]: Whoa, this is very *very* safe! Not sure this is really necessary for relatively small files like YAML configuration files, but hey, why not? 😅 Thank you ChatGPT, I guess! 😁

File: metricshub-agent/src/main/java/org/metricshub/web/service/ConfigurationFilesService.java [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+	 */
+	public void saveOrUpdateFile(final String fileName, final String content) {
+		final Path dir = requireConfigDir();
+		final Path target = resolveSafeYaml(dir, fileName);
+
+		try {
+			Files.createDirectories(dir);
+			// Write to a temp file, then move atomically
+			final Path tmp = Files.createTempFile(dir, fileName + ".", ".tmp");
+			Files.writeString(
```
Comment (bertysentry) [inline metricshub-agent/src/main/java/org/metricshub/web/service/ConfigurationFilesService.java:139]: Before saving the file, we should probably validate the file before saving it, and throw an exception if the file is invalid. If we save an invalid file, it will prevent the product from working.

Maybe we could add an option to skip the validation on save (`PUT /api/config-files/my-config.yaml?skipValidation=true`).

File: metricshub-agent/src/main/java/org/metricshub/web/service/ConfigurationFilesService.java [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+		final Path file = resolveSafeYaml(dir, fileName);
+
+		try {
+			if (!Files.deleteIfExists(file)) {
+				throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Configuration file not found.");
+			}
+		} catch (ResponseStatusException rse) {
+			throw rse;
+		} catch (IOException e) {
+			log.error("Failed to delete configuration file: '{}'. Error: {}", file, e.getMessage());
```
[the remark this answers is not in the document]
Comment (NassimBtk) [reply metricshub-agent/src/main/java/org/metricshub/web/service/ConfigurationFilesService.java:199]: Maybe using Log4J [PatternLayout](https://logging.apache.org/log4j/2.x/manual/pattern-layout.html) using `%throwable`  to be tested
